### ISSUE TITLE
Write an architecture decision records

### ISSUE THREAD

**Issue by Katolus** [Katolus/functions#132]
Write a document that would address the architecture of the projects touching on components, user flow, dependencies etc...

In addition to initial scope this issue should include work on the other `adrs` to update them to a final version.

This issues is strongly dependent on a whole bunch of documentation written prior to this.

**Comment by Katolus** [comment]
And implement it in the current or another issue.

**Comment by Katolus** [comment]
Depends on #148
